lib/freedom_patches/ember_compat_handlebars.rb

```diff
@@ -1,32 +1,47 @@
 # barber patches to re-route raw compilation via ember compat handlebars
-#
 
-module Barber
-  class EmberCompatPrecompiler < Barber::Precompiler
-
-    def sources
-      [File.open("#{Rails.root}/vendor/assets/javascripts/handlebars.js"), precompiler]
-    end
+class Barber::Precompiler
+  def sources
+    [File.open("#{Rails.root}/vendor/assets/javascripts/handlebars.js"), precompiler]
+  end
 
-    def precompiler
-    @precompiler ||= StringIO.new <<END
-      var Discourse = {};
-      #{File.read(Rails.root + "app/assets/javascripts/discourse/lib/ember_compat_handlebars.js")}
-      var Barber = {
-        precompile: function(string) {
-          return Discourse.EmberCompatHandlebars.precompile(string,false).toString();
-        }
-      };
+  def precompiler
+  @precompiler ||= StringIO.new <<END
+    var Discourse = {};
+    #{File.read(Rails.root + "app/assets/javascripts/discourse/lib/ember_compat_handlebars.js")}
+    var Barber = {
+      precompile: function(string) {
+        return Discourse.EmberCompatHandlebars.precompile(string,false).toString();
+      }
+    };
 END
+  end
+end
+
+module Discourse
+  module Ember
+    module Handlebars
+      module Helper
+        def precompile_handlebars(string)
+          "Discourse.EmberCompatHandlebars.template(#{Barber::Precompiler.compile(string)});"
+        end
+
+        def compile_handlebars(string)
+          "Discourse.EmberCompatHandlebars.compile(#{indent(string).inspect});"
+        end
+      end
     end
   end
 end
 
 class Ember::Handlebars::Template
-  def precompile_handlebars(string)
-    "Discourse.EmberCompatHandlebars.template(#{Barber::EmberCompatPrecompiler.compile(string)});"
-  end
-  def compile_handlebars(string)
-    "Discourse.EmberCompatHandlebars.compile(#{indent(string).inspect});"
+  include Discourse::Ember::Handlebars::Helper
+
+  # FIXME: Previously, ember-handlebars-templates uses the logical path which incorrectly
+  # returned paths with the `.raw` extension and our code is depending on the `.raw`
+  # to find the right template to use.
+  def actual_name(input)
+    actual_name = input[:name]
+    input[:filename].include?('.raw') ? "#{actual_name}.raw" : actual_name
```

Comment:
Yes those templates are ones that are passed to the raw handlebars renderer (ie, not Ember) for performance reasons. I'd be okay with a different naming convention if that makes it easier. For example, if the template name ends with `-raw`